sensors/trigger.go

```diff
@@ -30,60 +31,126 @@ import (
-				sec.log.Error().Err(err).Msg("failed to create K8s event to log trigger execution")
+			sn.MarkNodePhase(sec.sensor, trigger.Name, v1alpha1.NodeTypeTrigger, v1alpha1.NodePhaseError, nil, &sec.log, fmt.Sprintf("failed to execute trigger. err: %+v", err))
+
+			// escalate using K8s event
+			labels[common.LabelEventType] = string(common.EscalationEventType)
+			if err := common.GenerateK8sEvent(sec.kubeClient, fmt.Sprintf("failed to execute trigger %s", trigger.Name), common.EscalationEventType,
+				"trigger failure", sec.sensor.Name, sec.sensor.Namespace, sec.controllerInstanceID, sensor.Kind, labels); err != nil {
+				sec.log.Error().Err(err).Msg("failed to create K8s event to escalate trigger failure")
 			}
+			continue
 		}
 
-		// increment completion counter
-		sec.sensor.Status.CompletionCount = sec.sensor.Status.CompletionCount + 1
+		// mark trigger as complete.
+		sn.MarkNodePhase(sec.sensor, trigger.Name, v1alpha1.NodeTypeTrigger, v1alpha1.NodePhaseComplete, nil, &sec.log, "successfully executed trigger")
 
-		// create K8s event to mark the trigger round completion
 		labels[common.LabelEventType] = string(common.OperationSuccessEventType)
-		if err := common.GenerateK8sEvent(sec.kubeClient, fmt.Sprintf("completion count:%d", sec.sensor.Status.CompletionCount), common.OperationSuccessEventType,
-			"triggers execution round completion", sec.sensor.Name, sec.sensor.Namespace, sec.controllerInstanceID, sensor.Kind, labels); err != nil {
-			sec.log.Error().Err(err).Msg("failed to create K8s event to log trigger execution round completion")
+		if err := common.GenerateK8sEvent(sec.kubeClient, fmt.Sprintf("trigger %s executed successfully", trigger.Name), common.OperationSuccessEventType,
+			"trigger executed", sec.sensor.Name, sec.sensor.Namespace, sec.controllerInstanceID, sensor.Kind, labels); err != nil {
+			sec.log.Error().Err(err).Msg("failed to create K8s event to log trigger execution")
 		}
+	}
 
-		// Mark all signal nodes as active
-		for _, dep := range sec.sensor.Spec.Dependencies {
-			sn.MarkNodePhase(sec.sensor, dep.Name, v1alpha1.NodeTypeEventDependency, v1alpha1.NodePhaseActive, nil, &sec.log, "node is re-initialized")
-		}
-		return
+	// increment completion counter
+	sec.sensor.Status.CompletionCount = sec.sensor.Status.CompletionCount + 1
+
+	// create K8s event to mark the trigger round completion
+	labels[common.LabelEventType] = string(common.OperationSuccessEventType)
+	if err := common.GenerateK8sEvent(sec.kubeClient, fmt.Sprintf("completion count:%d", sec.sensor.Status.CompletionCount), common.OperationSuccessEventType,
+		"triggers execution round completion", sec.sensor.Name, sec.sensor.Namespace, sec.controllerInstanceID, sensor.Kind, labels); err != nil {
+		sec.log.Error().Err(err).Msg("failed to create K8s event to log trigger execution round completion")
+	}
+
+	// Mark all dependency nodes as active
```

Comment:
so every trigger execution resets the sensor nodes to `active` so that this sensor can be triggered again.?